beluga_tutorial/src/main.cpp

```diff
@@ -0,0 +1,278 @@
+    node["landmark_map"].push_back(lmp);
+  }
+
+  for (auto&& [cycle, data] : tutorial_dataset.sim_data | ranges::views::enumerate) {
+    node["simulation_dataset"][cycle]["ground_truth"] = data.ground_truth;
+
+    auto current = node["simulation_dataset"][cycle]["particles"]["current"];
+    current["states"] = beluga::views::states(data.current) | ranges::to<std::vector<double>>;
+    current["weights"] = beluga::views::weights(data.current) | ranges::to<std::vector<double>>;
+
+    auto propagate = node["simulation_dataset"][cycle]["particles"]["propagate"];
+    propagate["states"] = beluga::views::states(data.propagate) | ranges::to<std::vector<double>>;
+    propagate["weights"] = beluga::views::weights(data.propagate) | ranges::to<std::vector<double>>;
+
+    auto reweight = node["simulation_dataset"][cycle]["particles"]["reweight"];
+    reweight["states"] = beluga::views::states(data.reweight) | ranges::to<std::vector<double>>;
+    reweight["weights"] = beluga::views::weights(data.reweight) | ranges::to<std::vector<double>>;
+
+    auto resample = node["simulation_dataset"][cycle]["particles"]["resample"];
+    resample["states"] = beluga::views::states(data.resample) | ranges::to<std::vector<double>>;
+    resample["weights"] = beluga::views::weights(data.resample) | ranges::to<std::vector<double>>;
+
+    auto estimation = node["simulation_dataset"][cycle]["estimation"];
+    estimation["mean"] = std::get<0>(data.estimation);
+    estimation["sd"] = std::get<1>(data.estimation);
+  }
+
+  std::filesystem::create_directory(path.parent_path());
+  std::ofstream fout(path, std::ios::trunc);
+  fout << node;
+  fout.close();
+}
+
+int main(int argc, char* argv[]) {
+  if (argc < 3) {
+    std::cerr << "Usage: " << argv[0] << "<params_directory_path> <params_filename>\n";
+    return 1;
+  }
+
+  const std::string directory = argv[1];
+  const std::string filename = argv[2];
+  const auto parameters_file_path = std::filesystem::path{directory} / filename;
+
+  TutorialParams tutorial_params;
+  load_params_from_yaml(parameters_file_path, tutorial_params);
+
+  // Create the TutorialDataset object for record the simulation data
+  TutorialDataset tutorial_dataset;
+
+  LandmarkMapVector landmark_map;
+  load_landmark_map_from_yaml(std::filesystem::path{directory} / tutorial_params.landmark_map_file_name, landmark_map);
+  tutorial_dataset.landmark_map = landmark_map;
+
+  // Generate particles
+  std::normal_distribution<double> initial_distribution(tutorial_params.initial_pose, tutorial_params.initial_pose_sd);
+  auto particles = beluga::views::sample(initial_distribution) |                       //
+                   ranges::views::transform(beluga::make_from_state<Particle>) |       //
+                   ranges::views::take_exactly(tutorial_params.number_of_particles) |  //
+                   ranges::to<beluga::TupleVector>;
+
+  // Execute the particle filter using beluga
+  double current_pose{tutorial_params.initial_pose};
```

Comment:
Consider using "position" instead of "pose", which implies some directional element. Here and in other places.

Reply:
> Let's say that in the sign of the position, implicitly this is the direction to which it points.

I don't think that's correct. The position sign could be negative but that doesn't tell you anything about where it is pointing.

Reply:
Yes, you are right! I'm change the all the names from pose to position, thank!